### ISSUE TITLE
Global color picker UX improvment 

### ISSUE THREAD

**Issue by TurboGit** [darktable-org/darktable#17887]
**Is your feature request related to a problem? Please describe.**

Yes, the issue is that the colorpicker button in the global color-picker has kind of a three state.

- `Left-click` : point colorpicker
- `Right-click` : area colorpicker

And so to disable it one need to either `Left-click` if point color picker selected otherwise `Right-click`. This is really annoying.

**Describe the solution you'd like**

- `Left-click `: point colorpicker
- `Right-click` : area colorpicker

- `Left-click` / `Right-click` again would disable the picker

The question is now how to switch between point / area color picker ?

My proposal would be a `Ctrl+click` on the canvas would switch the mode from point / area.

EDIT:
Not sure about this last point and also all pickers are using the same UX. More thinking is needed here.

**Comment by jenshannoschwalm** [comment]
> Yes, the issue is that the colorpicker button in the global color-picker has kind of a three state.
> 
>     * `Left-click` : point colorpicker
> 
>     * `Right-click` : area colorpicker
> 
> 
> And so to disable it one need to either `Left-click` if point color picker selected otherwise `Right-click`. This is really annoying.

What about: If we `left-click` on a `point picker` it gets disabled, if we `right-click` on a `point picker` that gets disabled and an `area picker` is initialized? (plus similar behaviour for `area picker` being active right now.

**Comment by elstoc** [comment]
I would have it always be a simple toggle behaviour, as this gives good visual feedback and is more obvious: 
- If the picker is disabled, left/right click enables point/area picker respectively
- If the picker is enabled, both left and right click toggle it off

**Comment by todd-prior** [comment]
@elstoc I would agree as will with your comments and if we did have a single toggle and maybe even if not I still think it might be nice to add an icon down in the lower tray area where color assessment etc live.... esp for use cases where the picker is still active and you go to say do a perspective edit in rotate and perspective and you have to go over find the color picker module and then disable so that you can do the lines, I think the tone eq widget also wont show up for mouse wheel scrolling if the global picker is on and there are a few other things as well that you have to disable the picker for...it would be convenient to just have a toggle to hit at the bottom of the screen rather than have to navigate to the color picker module.... toggling it back on from that icon should just go back to the last selected mode.... Maybe it just me or I just make a short cut to do this...

**Comment by Zi7ar21** [comment]
Hey, while we're at it, can we make the color picker useful for colorimetric editing? Currently it only displays 8-bit values. It would be nice if this could pick HDR colors at multiple stages of the processing pipeline, so that it's clear what the values actually mean (what color space are they in?) and are actually useful. Thanks!

**Comment by elstoc** [comment]
@Zi7ar21 this sounds like a separate request and should be raised as a new issue. At the moment the global picker is at the end of the pipe and I'd guess it's hard (and would need some potentially nasty UI) to allow it to arbitrarily operate at different points (and in the corresponding colour spaces)

**Comment by jenshannoschwalm** [comment]
@TurboGit any plans on this?
